### PR TITLE
Update version to 0.10.0-beta.2

### PR DESCRIPTION
**Pull request by eeeebbbbrrrr**
This is the third beta in the pgrx v0.10.x series.  It contains a number of soundness fixes, better error handling, more testing, and other general code cleanup.

## Soundness Issues

* `AnyNumeric` is no longer backed by Postgres-allocated memory by @eeeebbbbrrrr in https://github.com/pgcentralfoundation/pgrx/pull/1216

## CI and general Testing Support

* Testing help by @eeeebbbbrrrr in https://github.com/pgcentralfoundation/pgrx/pull/1203
* Type testability cleanup by @eeeebbbbrrrr in https://github.com/pgcentralfoundation/pgrx/pull/1204
* Type roundtrip tests by @eeeebbbbrrrr in https://github.com/pgcentralfoundation/pgrx/pull/1185
* Stop SpiClient soundness from regressing by @workingjubilee in https://github.com/pgcentralfoundation/pgrx/pull/1214
* Initial valgrind support by @thomcc in https://github.com/pgcentralfoundation/pgrx/pull/1218
* Add a env flag that can be set to skip `#[pg_test]`-generated tests. by @thomcc in https://github.com/pgcentralfoundation/pgrx/pull/1239
* Ignores UI tests for MUSL environments by @BradyBonnette in https://github.com/pgcentralfoundation/pgrx/pull/1235
* Changes GHA workflows to use new upgraded runners by @BradyBonnette in https://github.com/pgcentralfoundation/pgrx/pull/1225

## General Improvements

* Add support for handling SIGINT and SIGCHLD from bgworker by @JelteF in https://github.com/pgcentralfoundation/pgrx/pull/1229
* Fix issue #1076: Properly handle dependency graph of `Result<T, _>` by @eeeebbbbrrrr in https://github.com/pgcentralfoundation/pgrx/pull/1241

## Improved Error Reporting

* Try to smartly propagate fs errors by @workingjubilee in https://github.com/pgcentralfoundation/pgrx/pull/1186
* Addresses cargo-pgrx error reporting by @BradyBonnette in https://github.com/pgcentralfoundation/pgrx/pull/1238
* Cleanup the error when cargo-pgrx version doesn't match Cargo.toml by @eeeebbbbrrrr in https://github.com/pgcentralfoundation/pgrx/pull/1240

## Additional Postgres Headers

* Add operator and cache related api by @VoVAllen in https://github.com/pgcentralfoundation/pgrx/pull/1242
* Add foreign table headers by @workingjubilee in https://github.com/pgcentralfoundation/pgrx/pull/1226
* Add postmaster related api by @JelteF in https://github.com/pgcentralfoundation/pgrx/pull/1237

## Internal Code Organization

* Modularize pgrx::spi by @workingjubilee in https://github.com/pgcentralfoundation/pgrx/pull/1219
* Modularize the interior of pgrx-pg-sys by @workingjubilee in https://github.com/pgcentralfoundation/pgrx/pull/1227

## Postgres 16-motivated Changes

* Add a workaround for the pg16/homebrew/icu4c situation  by @thomcc in https://github.com/pgcentralfoundation/pgrx/pull/1206

## General Project Stuff

* Add security policy by @johnrballard in https://github.com/pgcentralfoundation/pgrx/pull/1207

## New Contributors
* @johnrballard made their first contribution in https://github.com/pgcentralfoundation/pgrx/pull/1207
* @VoVAllen made their first contribution in https://github.com/pgcentralfoundation/pgrx/pull/1242

**Full Changelog**: https://github.com/pgcentralfoundation/pgrx/compare/v0.10.0-beta.1...v0.10.0-beta.2

### CODE REVIEW

File: cargo-pgrx/Cargo.toml [hunk 1]
```diff
@@ -25,19 +25,19 @@ edition = "2021"
 
 [dependencies]
 atty = "0.2.14"
-cargo_metadata = "0.15.4"
+cargo_metadata = "=0.15.4" # locked b/c we're using an older `clap`
```
Comment (workingjubilee) [inline cargo-pgrx/Cargo.toml:28]: we have a bunch of different brass instruments but no trombone emoji... sad.

**Review by workingjubilee — APPROVED** [review]

**Review by BradyBonnette — APPROVED** [review]
looks good!

**Review by thomcc — APPROVED** [review]

**Comment by JelteF** [comment]
No wait.

**Comment by JelteF** [comment]
Nvm. the patch that was only on master got added to develop too. LGTM too then.

File: cargo-pgrx/Cargo.toml (hunk 1 above)
Comment (eeeebbbbrrrr) [reply cargo-pgrx/Cargo.toml:28]: I own 3, Jubilee!  Plus a trumpet and a bugle.

**Comment by eeeebbbbrrrr** [comment]
> Nvm. the patch that was only on master got added to develop too. LGTM too then.

Yeah, that was on our radar.  ;)

File: cargo-pgrx/Cargo.toml (hunk 1 above)
Comment (eeeebbbbrrrr) [reply cargo-pgrx/Cargo.toml:28]: image

**Pull request merged**